azerothcore/azerothcore-wotlk
# fix(Scripts/AllCommandScript) CanExecuteCommand renamed and moved to be executed for all commands.

## Changes Proposed:

This PR proposes changes to:

When using the AllCommands::CanExecuteCommand hook, only failed commands are captured.
This is because the call is inside of the failure routine, so I moved it.

## Issues Addressed:

None

## Tests Performed:

This PR has been:


## How to Test the Changes:

1. Set a breakpoint on line 328 of ChatCommand.cpp
2. Observe it is hit for all commands, failed or not.
3. Yay?

Alternatively you can download [this](https://github.com/AnchyDev/ButWhy) module and observe that only failed commands are blocked.

## Known Issues and TODO List:

I don't see any issue with capturing all commands.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
